File: Irrgarten_Ruby/Irrgarten_Ruby_P4/Monster.rb
#encoding:utf-8

require_relative 'Dice'
require_relative 'LabyrinthCharacter'

module Irrgarten 
	class Monster < LabyrinthCharacter
		@@INITIAL_HEALTH = 5
		
		
		def initialize ( name, intelligence, strength) #string,float,float
			super(name,intelligence,strength,@@INITIAL_HEALTH)
		end
		
		def attack()
			return Dice.intensity(@strength)
		end
		
		def defend( received_attack ) #float
			is_dead = dead
			
			if (!is_dead)
				defensive_energy = Dice.intensity(@intelligence)
				if (defensive_energy < received_attack)
					got_wounded
					is_dead = dead
				end
			end
			
			is_dead
		end	
	end
end

File: Irrgarten_Java/Irrgarten_P5/src/irrgarten/Monster.java

package irrgarten;

/**
 *
 * @author leonin04
 */
public class Monster extends LabyrinthCharacter {
    private static final int INITIAL_HEALTH = 5;
    
    
    public Monster (String name, float intelligence, float strength){
        super(name,intelligence,strength,INITIAL_HEALTH);
    }
    
    @Override
    public float attack(){
        return Dice.intensity(super.getStrength());
    }
    
    @Override
    public boolean defend(float receivedAttack){
        boolean isDead = dead();
        if(isDead == false){
            float defensiveEnergy = Dice.intensity(super.getIntelligence());
            if (defensiveEnergy < receivedAttack){
                super.getWounded();
                isDead = dead();
            }
            
        }
        return isDead;
    }  
    
    @Override
    public String toString(){
        return super.toString()+"\n";
    }
    
}
